### PR TITLE
Add optional header parameter to the MeetingPOSTLogger

### PR DESCRIPTION
**Pull request by RidipDe**
**Description of changes:**
Add optional header parameter to the MeetingPOSTLogger

**Testing**
Unit tests

1. Have you successfully run `npm run build:release` locally?
Yes
2. How did you test these changes?
Unit tests
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?
Yes, meeting
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
Yes - added an optional parameter
Yes.
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/logger/MeetingSessionPOSTLogger.ts [hunk 1]
```diff
@@ -27,7 +27,8 @@ export default class MeetingSessionPOSTLogger implements Destroyable {
     private batchSize: number,
     private intervalMs: number,
     private url: string,
-    private level = LogLevel.WARN
+    private level = LogLevel.WARN,
+    private headers?: Record<string, string>
```
Comment (simmkyu) [inline src/logger/MeetingSessionPOSTLogger.ts:31]: In TypeScript DOM types, `headers` is of the `HeadersInit` type.
https://github.com/microsoft/TypeScript/blob/c671fe1bc2878e1ba60f4199856d1a57c3cc99ff/lib/lib.dom.d.ts#L1618

```
type HeadersInit = Headers | string[][] | Record<string, string>;
```

File: src/logger/MeetingSessionPOSTLogger.ts [hunk 2]
```diff
@@ -102,10 +103,16 @@ export default class MeetingSessionPOSTLogger implements Destroyable {
       const batch = this.logCapture.slice(0, batchSize);
       const body = this.makeRequestBody(batch);
       try {
-        const response = await fetch(this.url, {
-          method: 'POST',
-          body,
-        });
+        const response = this.headers
```
Comment (simmkyu) [inline src/logger/MeetingSessionPOSTLogger.ts:106]: tip: You can write the code using one `fetch`.

```
const response = await fetch(this.url, {
  method: 'POST',
  body,
  ...(this.headers ? {
    headers: this.headers
  } : {})
});
```

**Review by simmkyu — COMMENTED** [review]
You need additional tests for `Headers` and `string[][]`. See my comment in the post logger.

File: src/logger/MeetingSessionPOSTLogger.ts (hunk 1 above)
Comment (RidipDe) [reply src/logger/MeetingSessionPOSTLogger.ts:31]: Discussed with Kyu offline, based on the existing pattern in the JS SDK https://github.com/aws/amazon-chime-sdk-js/blob/6596622238e69c5202c3e697aeeedd54f2a0d35b/libs/voicefocus/types.d.ts#L125 we will follow only the Record<string, string> type as of now.

File: src/logger/MeetingSessionPOSTLogger.ts (hunk 2 above)
Comment (RidipDe) [reply src/logger/MeetingSessionPOSTLogger.ts]: Thanks, great to know!
Updated to this format.

File: package.json [hunk 3]
```diff
@@ -79,6 +79,7 @@
   "dependencies": {
     "@types/dom-mediacapture-record": "^1.0.7",
     "@types/ua-parser-js": "^0.7.35",
+    "aws-embedded-metrics": "^2.0.4",
```
Comment (simmkyu) [inline package.json:82]: This dependency doesn't seem to be related to this PR?

File: src/logger/MeetingSessionPOSTLogger.ts [hunk 4]
```diff
@@ -105,6 +106,11 @@ export default class MeetingSessionPOSTLogger implements Destroyable {
         const response = await fetch(this.url, {
           method: 'POST',
           body,
+          ...(this.headers
```
Comment (simmkyu) [inline src/logger/MeetingSessionPOSTLogger.ts:109]: nit: for readability.. Ignore this commit if the Prettier has automatically fixed your code.

```
...(this.headers ? {
  headers: this.headers
} : {}),
```

File: package.json (hunk 3 above)
Comment (RidipDe) [reply package.json]: Hmm, thats weird. Have removed it.

File: src/logger/MeetingSessionPOSTLogger.ts (hunk 4 above)
Comment (RidipDe) [reply src/logger/MeetingSessionPOSTLogger.ts:109]: npm run build:release is fixing that

**Review by simmkyu — APPROVED** [review]
